### PR TITLE
fix `gr_read`  panic when arguments are incorrect

### PR DESCRIPTION
**Pull request by grishasobol**

### CODE REVIEW

File: core-backend/sandbox/src/funcs.rs [hunk 1]
```diff
@@ -310,10 +308,15 @@ where
 
         let mut f = || {
             let msg = ctx.ext.msg().to_vec();
+            if at > msg.len() || at + len > msg.len() {
```
Comment (breathx) [inline core-backend/sandbox/src/funcs.rs:311]:
```suggestion
            if at + len > msg.len() {
```
First check is automatically included in second one, cause if `at > msg.len()` than with any `len` `at + len > msg.len()`

Same thing in `wasmi` backend

**Review by breathx — APPROVED** [review]
Great catch!

File: core-backend/sandbox/src/funcs.rs (hunk 1 above)
Comment (grishasobol) [reply core-backend/sandbox/src/funcs.rs:311]: Yes, but we must make then checked add or saturated. Ok will do this

**Review by ukint-vs — APPROVED** [review]

File: core-backend/sandbox/src/funcs.rs (hunk 1 above)
Comment (breathx) [reply core-backend/sandbox/src/funcs.rs:311]: > Yes, but we must make then checked add or saturated. Ok will do this

I prefer saturating math here

Comment (breathx) [reply core-backend/sandbox/src/funcs.rs:311]: Btw, it must be saturated even with the previous condition statement

Comment (grishasobol) [reply core-backend/sandbox/src/funcs.rs]: Done

**Pull request merged**